### PR TITLE
🧹 Removing the component exports for `CustomFieldExtensions` in `plugin-scaffolder`

### PR DESCRIPTION
**Pull request by benjdlambert**
#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/scaffolder/api-report.md [hunk 1]
```diff
@@ -28,6 +28,7 @@ import { ScmIntegrationRegistry } from '@backstage/integration';
 import { TaskSpec } from '@backstage/plugin-scaffolder-common';
 import { TemplateEntityV1beta2 } from '@backstage/plugin-scaffolder-common';
 
+// Warning: (ae-forgotten-export) The symbol "FieldExtensionComponent" needs to be exported by the entry point index.d.ts
```
Comment (Rugvip) [inline plugins/scaffolder/api-report.md:31]: 👁️ _ 👁️

**Review by Rugvip — COMMENTED** [review]
🎉

File: .changeset/swift-roses-hug.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-scaffolder': patch
```
Comment (Rugvip) [inline .changeset/swift-roses-hug.md:2]: `minor` 😅

**Review by Rugvip — APPROVED** [review]
Nice :grin:

pre-approved, but should fix the changeset before merj

**Comment by Crevil** [comment]
This change removed the export of `EntityPicker` which we used for creating custom extensions.

I'm having a bit of a hard time figuring out how we can do this now? What we have previously been doing was to create our own `DomainPicker` implemented like the existing  pickers like `OwnedEntityPicker`.

Can I somehow use `EntityPickerFieldExtension` to create a variant?

**Pull request merged**